### PR TITLE
Add Interop 2024 dashboard

### PR DESCRIPTION
**Pull request by DanielRyanSmith**
Adds the 2024 Interop Dashboard and category data.

The change for separating the Chrome and Edge scores will be in a subsequent PR.

This PR does NOT set the 2024 dashboard as the default dashboard, and Interop 2024 will not be visible to users by landing this change. Adding Interop 2024 to the as the default route for the Interop dashboard will be part of a subsequent PR.

Screenshot 2024-01-19 at 7 43 42 PM
Screenshot 2024-01-19 at 7 43 52 PM

### CODE REVIEW

File: webapp/components/interop-data.js [hunk 1]
```diff
@@ -579,5 +579,408 @@ export const interopData = {
...
+        'url': 'https://github.com/web-platform-tests/interop-2023-mobile-testing',
+        'scores_over_time': []
+      },
+      {
+        'name': 'WebAssembly Testing',
+        'url': '',
+        'scores_over_time': []
+      },
+    ],
+    'investigation_weight': 0.0,
```
Comment (foolip) [inline webapp/components/interop-data.js:667]: Can we start omitting this? It was only for Interop 2022. Perhaps the code that uses `investigation_weight` can just tolerate it being missing and not mix in scores?

File: webapp/components/interop-data.js [hunk 2]
```diff
@@ -579,5 +579,408 @@ export const interopData = {
...
+      },
+      'interop-2022-viewport': {
+        'description': 'Viewport Units',
+        'mdn': '',
+        'spec': 'https://drafts.csswg.org/css-values/#viewport-relative-units',
+        'tests': '/results/css/css-values?label=master&product=chrome&product=firefox&product=safari&aligned&view=interop&q=label%3Ainterop-2022-viewport',
+        'countsTowardScore': false
+      },
+      'interop-2022-webcompat': {
+        'description': 'Web Compat',
```
Comment (foolip) [inline webapp/components/interop-data.js:719]: As in https://github.com/web-platform-tests/wpt.fyi/pull/3653:

```suggestion
        'description': 'Web Compat 2022',
```

File: webapp/components/interop-data.js [hunk 3]
```diff
@@ -579,5 +579,408 @@ export const interopData = {
...
+      },
+      'interop-2024-dir': {
+        'description': 'Direction',
+        'mdn': '',
+        'spec': '',
+        'tests': '/results/html/dom/elements/global-attributes?label=master&label=experimental&product=chrome&product=firefox&product=safari&aligned&view=interop&q=label%3Ainterop-2024-dir',
+        'countsTowardScore': true
+      },
+      'interop-2024-font-size-adjust': {
+        'description': 'CSS font-size-adjust',
```
Comment (foolip) [inline webapp/components/interop-data.js:754]:
```suggestion
        'description': 'font-size-adjust',
```

File: webapp/components/interop-data.js [hunk 4]
```diff
@@ -579,5 +579,408 @@ export const interopData = {
...
+      },
+      'interop-2024-scrollbar': {
+        'description': 'Scrollbar styling',
+        'mdn': '',
+        'spec': '',
+        'tests': '/results/css?label=master&label=experimental&product=chrome&product=firefox&product=safari&aligned&view=interop&q=label%3Ainterop-2024-scrollbar',
+        'countsTowardScore': true
+      },
+      'interop-2024-text-wrap': {
+        'description': 'CSS text-wrap: balance',
```
Comment (foolip) [inline webapp/components/interop-data.js:817]:
```suggestion
        'description': 'text-wrap: balance',
```

File: webapp/components/interop-data.js [hunk 5]
```diff
@@ -579,5 +579,408 @@ export const interopData = {
...
+      },
+      'interop-2024-video-rvfc': {
+        'description': 'requestVideoFrameCallback',
+        'mdn': '',
+        'spec': '',
+        'tests': '/results/video-rvfc?label=experimental&label=master&product=chrome&product=firefox&product=safari&aligned&view=interop&q=label%3Ainterop-2024-video-rvfc',
+        'countsTowardScore': true
+      },
+      'interop-2024-scrollbar': {
+        'description': 'Scrollbar styling',
```
Comment (foolip) [inline webapp/components/interop-data.js:810]:
```suggestion
        'description': 'Scrollbar Styling',
```

File: webapp/components/interop-data.js (hunk 1 above)
Comment (DanielRyanSmith) [reply webapp/components/interop-data.js:667]: I can submit a separate PR to remove this from non-relevant years 🙂

File: webapp/components/interop-dashboard.js [hunk 6]
```diff
@@ -629,12 +629,13 @@ class InteropDashboard extends PolymerElement {
...
     this.$.toggleExperimental.setAttribute('aria-pressed', !this.stable);
     // Keep the block-level design for interop 2021-2022
-    if (this.year !== '2023') {
+    if (this.year === '2021' || this.year === '2022') {
       const gridContainerDiv = this.shadowRoot.querySelector('.grid-container');
       gridContainerDiv.style.display = 'block';
       gridContainerDiv.style.width = '700px';
       gridContainerDiv.style.margin = 'auto';
-      // 2023 also displays a special description which is not displayed in previous years.
+      // dashboards after 2022 also display a special description
```
Comment (KyleJu) [inline webapp/components/interop-dashboard.js:637]:
```suggestion
      // Dashboards after 2022 also display a special description,
```

**Review by KyleJu — APPROVED** [review]

**Pull request merged**